qadence/serialization.py

```diff
@@ -43,27 +47,127 @@ def file_extension(file: Path | str) -> str:
     AbstractBlock,
     QuantumCircuit,
     QuantumModel,
-    QNN,
-    TransformedModule,
     Register,
-    Basic,
+    core.Basic,
     torch.nn.Module,
 ]
 SUPPORTED_TYPES = TypingUnion[
     AbstractBlock,
     QuantumCircuit,
     QuantumModel,
-    QNN,
-    TransformedModule,
     Register,
-    Basic,
+    core.Basic,
     torch.nn.Module,
 ]
 
+EXPR_TYPES = TypingUnion[core.Expr, float, int, str, tuple, dict]
+
 
 ALL_BLOCK_NAMES = [
     n for n in dir(qadenceblocks) if not (n.startswith("__") and n.endswith("__"))
 ] + [n for n in dir(operations) if not (n.startswith("__") and n.endswith("__"))]
+SYMPY_EXPRS = [n for n in dir(core) if not (n.startswith("__") and n.endswith("__"))]
+QADENCE_PARAMETERS = [n for n in dir(parameters) if not (n.startswith("__") and n.endswith("__"))]
+
+
+THIS_PATH = os.path.dirname(__file__)
+GRAMMAR_FILE = os.path.join(THIS_PATH, "serial_expr_grammar.peg")
+
+
+def _parser_fn() -> ParserPEG:
+    with open(GRAMMAR_FILE, "r") as f:
+        grammar = f.read()
+    return ParserPEG(grammar, "Program")
+
+
+def parse_expr_fn(code: str) -> bool:
+    parser = _parser_fn()
+    try:
+        parser.parse(code)
+    except NoMatch as err:
+        print(f"no match error: {err}")
+        return False
+    else:
+        return True
+
+
+@dataclass
+class SerialModel:
+    d: dict = dataclass_field(default_factory=dict)
+    value: Any = dataclass_field(init=False)
+
+
+@dataclass
+class BlockTypeSerial(SerialModel):
+    value: AbstractBlock = dataclass_field(init=False)
```

Comment:
Is it coherent to initialise to `False` ? I mean from a typing point of view.

Reply:
Oh, I see! `init=False` is just saying that I don't want this attribute to be accessible as a parameter in the `__init__`
like:
```python
@dataclass
class X1:
    a: str
    b: str = field(init=False)
```
is equivalent to

```python
class X2:
    def __init__(self, a: str):
        self.a = a
        self.b: str
```